modules/dreamview/backend/perception_camera_updater/perception_camera_updater.cc

```diff
@@ -0,0 +1,221 @@
+  }
+  enabled_ = false;
+}
+
+void PerceptionCameraUpdater::GetImageLocalization(
+    std::vector<double> *localization) {
+  double timestamp_diff = DBL_MAX;
+  auto iter = localization_queue_.rbegin();
+  Pose image_pos;
+  for (; iter != localization_queue_.rend(); ++iter) {
+    const double tmp_diff = (*iter)->measurement_time()
+        - current_image_timestamp_;
+    if (tmp_diff > 0) {
+      timestamp_diff = tmp_diff;
+      image_pos = (*iter)->pose();
+    } else {
+      if (std::abs(tmp_diff) < timestamp_diff) {
+        image_pos = (*iter)->pose();
+      }
+      break;
+    }
+  }
+  // clean useless localization
+  auto iter_begin = localization_queue_.begin();
+  while ((*iter_begin) != (*iter)) {
+    iter_begin = localization_queue_.erase(iter_begin);
+  }
+
+  Eigen::Matrix4d localization_matrix;
+  localization_matrix.setConstant(0);
+  Eigen::Quaterniond q;
+  q.x() = image_pos.orientation().qx();
+  q.y() = image_pos.orientation().qy();
+  q.z() = image_pos.orientation().qz();
+  q.w() = image_pos.orientation().qw();
+  localization_matrix.block<3, 3>(0, 0) = q.normalized().toRotationMatrix();
+  localization_matrix(0, 3) = image_pos.position().x();
+  localization_matrix(1, 3) = image_pos.position().y();
+  localization_matrix(2, 3) = image_pos.position().z();
+  localization_matrix(3, 3) = 1;
+
+  ConvertMatrixToArray(localization_matrix, localization);
+}
+
+void PerceptionCameraUpdater::OnImage(
+    const std::shared_ptr<CompressedImage> &compressed_image) {
+  if (!enabled_
+      || compressed_image->format() == "h265" /* skip video format */) {
+    return;
+  }
+
+  std::lock_guard<std::mutex> lock(image_mutex_);
+  if (compressed_image->has_measurement_time()) {
+    current_image_timestamp_ = compressed_image->measurement_time();
+  } else {
+    current_image_timestamp_ = compressed_image->header().timestamp_sec();
+  }
+
+  std::vector<uint8_t> compressed_raw_data(compressed_image->data().begin(),
+                                           compressed_image->data().end());
+  cv::Mat mat_image = cv::imdecode(compressed_raw_data, CV_LOAD_IMAGE_COLOR);
+  // Scale down image size properly to reduce data transfer latency through
+  // websocket and ensure image quality is acceptable meanwhile
+  cv::resize(
+      mat_image, mat_image,
+      cv::Size(static_cast<int>(mat_image.cols * kImageScale),
+               static_cast<int>(mat_image.rows * kImageScale)),
+      0, 0, CV_INTER_LINEAR);
+  std::vector<uint8_t> tmp_buffer;
+  cv::imencode(".jpg", mat_image, tmp_buffer, std::vector<int>() /* params */);
+
+  camera_update_.set_data(&(tmp_buffer[0]), tmp_buffer.size());
+}
+
+void PerceptionCameraUpdater::OnLocalization(
+    const std::shared_ptr<LocalizationEstimate> &localization) {
+  if (!enabled_) {
+    return;
+  }
+
+  std::lock_guard<std::mutex> lock(localization_mutex_);
+  localization_queue_.push_back(localization);
+}
+
+void PerceptionCameraUpdater::OnStaticTransform(
```

Comment:
is there any code source for handling the static transform? if yes, maybe we can state in the comment for reference